app/lib/account/theme/theme_settings.dart

```diff
@@ -121,20 +121,20 @@ class ThemeSettings extends ChangeNotifier {
     ));
   }
 
-  late VisualDensity _visualDensity;
-  VisualDensity get visualDensity => _visualDensity;
-  set visualDensity(VisualDensity value) {
-    _visualDensity = value;
+  late VisualDensitySetting _visualDensitySetting;
+  VisualDensitySetting get visualDensitySetting => _visualDensitySetting;
+  set visualDensitySetting(VisualDensitySetting value) {
+    _visualDensitySetting = value;
     notifyListeners();
 
     _keyValueStore.setString(currentVisualDensityCacheKey, value.serialize());
     _analytics.log(NamedAnalyticsEvent(
       name: 'ui_visual_density_changed',
       data: {
-        'visual_density': {
-          'horizontal': value.horizontal,
-          'vertical': value.vertical
-        }
+        // bool is not allowed in firebase_analytics so we use `toString`.
+        'isAdaptivePlatformDensity': value.isAdaptivePlatformDensity.toString(),
+        'horizontal': value.visualDensity.horizontal,
+        'vertical': value.visualDensity.vertical
```

Comment:
The old code would have always thrown an error here since the map inside map (`'data:': { 'visual_density': {...}}`) is not allowed by `firebase_analytics`.